diff --exclude-from=exclude -N -u -r nsalibselinux/include/selinux/selinux.h libselinux-1.17.12/include/selinux/selinux.h
--- nsalibselinux/include/selinux/selinux.h	2004-09-14 09:31:35.000000000 -0400
+++ libselinux-1.17.12/include/selinux/selinux.h	2004-09-15 16:36:13.000000000 -0400
@@ -197,6 +197,7 @@
    policy root directory. */
 extern const char *selinux_binary_policy_path(void);
 extern const char *selinux_failsafe_context_path(void);
+extern const char *selinux_removable_context_path(void);
 extern const char *selinux_default_context_path(void);
 extern const char *selinux_user_contexts_path(void);
 extern const char *selinux_file_context_path(void);
diff --exclude-from=exclude -N -u -r nsalibselinux/src/compat_file_path.h libselinux-1.17.12/src/compat_file_path.h
--- nsalibselinux/src/compat_file_path.h	2004-09-14 09:31:35.000000000 -0400
+++ libselinux-1.17.12/src/compat_file_path.h	2004-09-15 16:41:01.000000000 -0400
@@ -8,3 +8,5 @@
 S_(DEFAULT_TYPE, SECURITYDIR "/default_type")
 S_(BOOLEANS, SECURITYDIR "/booleans")
 S_(MEDIA_CONTEXTS, SECURITYDIR "/default_media")
+S_(REMOVABLE_CONTEXT, SECURITYDIR "/removable_context")
+
diff --exclude-from=exclude -N -u -r nsalibselinux/src/file_path_suffixes.h libselinux-1.17.12/src/file_path_suffixes.h
--- nsalibselinux/src/file_path_suffixes.h	2004-09-14 09:31:35.000000000 -0400
+++ libselinux-1.17.12/src/file_path_suffixes.h	2004-09-15 16:38:30.000000000 -0400
@@ -8,3 +8,4 @@
 S_(DEFAULT_TYPE, "/contexts/default_type")
 S_(BOOLEANS, "/booleans")
 S_(MEDIA_CONTEXTS, "/contexts/files/media")
+S_(REMOVABLE_CONTEXT, "/contexts/removable_context")
diff --exclude-from=exclude -N -u -r nsalibselinux/src/selinux_config.c libselinux-1.17.12/src/selinux_config.c
--- nsalibselinux/src/selinux_config.c	2004-09-14 09:31:36.000000000 -0400
+++ libselinux-1.17.12/src/selinux_config.c	2004-09-15 16:39:02.000000000 -0400
@@ -25,7 +25,8 @@
 #define DEFAULT_TYPE      6
 #define BOOLEANS          7
 #define MEDIA_CONTEXTS    8
-#define NEL               9
+#define REMOVABLE_CONTEXT 9
+#define NEL               10
 
 /* New layout is relative to SELINUXDIR/policytype. */
 static char *file_paths[NEL];
@@ -192,6 +193,11 @@
 }
 hidden_def(selinux_failsafe_context_path)
 
+const char *selinux_removable_context_path() {
+  return get_path(REMOVABLE_CONTEXT);
+}
+hidden_def(selinux_removable_context_path)
+
 const char *selinux_binary_policy_path() {
   return get_path(BINPOLICY);
 }
diff --exclude-from=exclude -N -u -r nsalibselinux/src/selinux_internal.h libselinux-1.17.12/src/selinux_internal.h
--- nsalibselinux/src/selinux_internal.h	2004-08-25 08:44:17.000000000 -0400
+++ libselinux-1.17.12/src/selinux_internal.h	2004-09-15 16:32:14.000000000 -0400
@@ -16,6 +16,7 @@
 hidden_proto(security_getenforce)
 hidden_proto(selinux_default_context_path)
 hidden_proto(selinux_failsafe_context_path)
+hidden_proto(selinux_removable_context_path)
 hidden_proto(selinux_file_context_path)
 hidden_proto(selinux_user_contexts_path)
 hidden_proto(selinux_booleans_path)
