--- libselinux-2.0.65/include/selinux/av_permissions.h.orig	2008-02-21 17:11:33.000000000 +0100
+++ libselinux-2.0.65/include/selinux/av_permissions.h	2008-09-10 22:17:10.876983118 +0200
@@ -542,6 +542,7 @@
 #define CAPABILITY__SETFCAP                       0x80000000UL
 #define CAPABILITY2__MAC_OVERRIDE                 0x00000001UL
 #define CAPABILITY2__MAC_ADMIN                    0x00000002UL
+#define CAPABILITY2__CONTEXT                      0x00000004UL
 #define PASSWD__PASSWD                            0x00000001UL
 #define PASSWD__CHFN                              0x00000002UL
 #define PASSWD__CHSH                              0x00000004UL
