--- libselinux-1.34.0/include/selinux/av_permissions.h.orig	2006-11-20 18:25:30.000000000 +0100
+++ libselinux-1.34.0/include/selinux/av_permissions.h	2007-01-28 00:30:57.517480000 +0100
@@ -525,6 +525,7 @@
 #define CAPABILITY__LEASE                         0x10000000UL
 #define CAPABILITY__AUDIT_WRITE                   0x20000000UL
 #define CAPABILITY__AUDIT_CONTROL                 0x40000000UL
+#define CAPABILITY__CONTEXT                       0x80000000UL
 #define PASSWD__PASSWD                            0x00000001UL
 #define PASSWD__CHFN                              0x00000002UL
 #define PASSWD__CHSH                              0x00000004UL
