--- libselinux/src/av_perm_to_string.h	2006-03-20 05:53:29.000000000 +0000
+++ libselinux/src/av_perm_to_string.h	2007-01-12 02:06:17.278580000 +0000
@@ -123,6 +123,7 @@
     S_(SECCLASS_CAPABILITY, CAPABILITY__LEASE, "lease")
     S_(SECCLASS_CAPABILITY, CAPABILITY__AUDIT_WRITE, "audit_write")
     S_(SECCLASS_CAPABILITY, CAPABILITY__AUDIT_CONTROL, "audit_control")
+    S_(SECCLASS_CAPABILITY, CAPABILITY__CONTEXT, "vserver_context")
     S_(SECCLASS_PASSWD, PASSWD__PASSWD, "passwd")
     S_(SECCLASS_PASSWD, PASSWD__CHFN, "chfn")
     S_(SECCLASS_PASSWD, PASSWD__CHSH, "chsh")
--- libselinux/include/selinux/av_permissions.h	2006-03-20 05:53:29.000000000 +0000
+++ libselinux/include/selinux/av_permissions.h	2007-01-12 02:07:24.624890000 +0000
@@ -565,6 +565,7 @@
 #define CAPABILITY__LEASE                         0x10000000UL
 #define CAPABILITY__AUDIT_WRITE                   0x20000000UL
 #define CAPABILITY__AUDIT_CONTROL                 0x40000000UL
+#define CAPABILITY__CONTEXT                       0x80000000UL
 
 #define PASSWD__PASSWD                            0x00000001UL
 #define PASSWD__CHFN                              0x00000002UL
--- libselinux-1.32./src/av_perm_to_string.h	2007-01-12 03:32:09.413722000 +0100
+++ libselinux-1.32/src/av_perm_to_string.h	2007-01-12 03:29:48.363722000 +0100
