--- libselinux-1.17.13/src/dso.h.orig	2004-09-26 16:07:37.528350528 +0200
+++ libselinux-1.17.13/src/dso.h	2004-09-26 16:08:16.827376168 +0200
@@ -7,8 +7,13 @@
 # define __hidden_proto(fct, internal)	\
      extern __typeof (fct) internal;	\
      extern __typeof (fct) fct __asm (#internal) hidden;
+#ifdef __alpha__
+# define hidden_def(fct) \
+     asm (".globl " #fct "\n" #fct " = " #fct "_internal");
+#else
 # define hidden_def(fct) \
      asm (".globl " #fct "\n.set " #fct ", " #fct "_internal");
+#endif
 #else
 # define hidden
 # define hidden_proto(fct)
