diff -Nur libselinux-1.0/include/Makefile libselinux-1.0.new/include/Makefile
--- libselinux-1.0/include/Makefile	2003-06-02 22:17:25.000000000 +0200
+++ libselinux-1.0.new/include/Makefile	2003-06-18 13:03:25.000000000 +0200
@@ -3,5 +3,5 @@
 INCDIR ?= $(PREFIX)/include/selinux
 
 install:
-	test -d $(INCDIR) || install -m 755 -o root -g root -d $(INCDIR)
-	install -m 644 -o root -g root $(wildcard selinux/*.h) $(INCDIR)
+	test -d $(INCDIR) || install -m 755 -d $(INCDIR)
+	install -m 644 $(wildcard selinux/*.h) $(INCDIR)
diff -Nur libselinux-1.0/src/Makefile libselinux-1.0.new/src/Makefile
--- libselinux-1.0/src/Makefile	2003-06-09 21:22:57.000000000 +0200
+++ libselinux-1.0.new/src/Makefile	2003-06-18 13:06:34.000000000 +0200
@@ -25,11 +25,10 @@
 	$(CC) -o $@ -c $(CFLAGS) $<
 
 install: all
-	test -d $(LIBDIR) || install -m 755 -o root -g root -d $(LIBDIR)
-	install -m 644 -o root -g root $(LIBA) $(LIBDIR)
-	install -m 755 -o root -g root $(LIBSO) $(LIBDIR)
+	test -d $(LIBDIR) || install -m 755 -d $(LIBDIR)
+	install -m 644 $(LIBA) $(LIBDIR)
+	install -m 755 $(LIBSO) $(LIBDIR)
 	cd $(LIBDIR) && ln -sf $(LIBSO) $(TARGET)
-	/sbin/ldconfig
 
 clean: 
 	rm -f $(OBJS) $(LIBA) $(LIBSO) $(TARGET) 
