--- libselinux-3.11/src/Makefile.orig	2026-07-01 18:51:05.000000000 +0200
+++ libselinux-3.11/src/Makefile	2026-07-13 20:14:14.543941657 +0200
@@ -196,7 +196,7 @@
 endif
 
 install-pywrap: pywrap
-	$(PYTHON) -m pip install --no-build-isolation --force-reinstall --prefix=$(PREFIX) `test -n "$(DESTDIR)" && echo --root $(DESTDIR) --ignore-installed --no-deps` dist/*.whl
+	$(PYTHON) -m installer --prefix=$(PREFIX) --compile-bytecode=0 --compile-bytecode=1 --compile-bytecode=2 `test -n "$(DESTDIR)" && echo --destdir=$(DESTDIR)` dist/*.whl
 	install -m 644 $(SWIGPYOUT) $(DESTDIR)$(PYTHONLIBDIR)/selinux/__init__.py
 	ln -sf --relative $(DESTDIR)$(PYTHONLIBDIR)/selinux/_selinux$(PYCEXT) $(DESTDIR)$(PYTHONLIBDIR)/_selinux$(PYCEXT)
 
